### PR TITLE
[CHORE] Simplify server version comparison

### PR DESCRIPTION
**Pull request by gerzonc**
## Proposed changes

Sometimes we have to test server version depending on the feature availability. For that, we use `semver` lib. Server version might have a suffix for development, like `-develop` or `-rc`, which breaks the default of semantic versioning. We have `coerce` to transform it back to `semver ` (e.g. `3.11.0-develop` to `3.11.0`). Otherwise, lower than or greater than won't work. The problem is that we have all of these conditions spread throughout the code.

In order to refactor this in a more simple way I just created two utility functions which work just as `lt` (lower than) and `gte` (greater than) but including `coerce` so we don't have to call it every time we want to compare server versions.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/lib/utils.js [hunk 1]
```diff
@@ -7,3 +9,7 @@ export const formatAttachmentUrl = (attachmentUrl, userId, token, server) => {
 	}
 	return encodeURI(`${ server }${ attachmentUrl }?rc_uid=${ userId }&rc_token=${ token }`);
 };
+
+export const lt = (currentServerVersion, oldServerVersion) => lowerThan(coerce(currentServerVersion), oldServerVersion);
```
Comment (diegolmello) [inline app/lib/utils.js:13]: Currently, we're using the functions from `semver` directly, but that's not meaninful.
Can you rename `lt` to `isServerVersionLowerThan`?
Same for `gte`.

@gerzonc

Comment (gerzonc) [reply app/lib/utils.js:13]: Alright

File: app/lib/methods/enterpriseModules.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ export function getEnterpriseModules() {
 	return new Promise(async(resolve) => {
 		try {
 			const { version: serverVersion, server: serverId } = reduxStore.getState().server;
-			if (serverVersion && gte(serverVersion, '3.1.0')) {
+			if (serverVersion && isServerVersionGreaterThan(serverVersion, '3.1.0')) {
```
Comment (diegolmello) [inline app/lib/methods/enterpriseModules.js:36]: @gerzonc One more thing here.
We're repeating ourselves on `serverVersion && ` in all calls.
Do you agree that should be inside of the util functions? :)

File: app/lib/utils.js [hunk 3]
```diff
@@ -7,3 +9,7 @@ export const formatAttachmentUrl = (attachmentUrl, userId, token, server) => {
 	}
 	return encodeURI(`${ server }${ attachmentUrl }?rc_uid=${ userId }&rc_token=${ token }`);
 };
+
+export const isServerVersionLowerThan = (currentServerVersion, oldServerVersion) => currentServerVersion && lt(coerce(currentServerVersion), oldServerVersion);
+
+export const isServerVersionGreaterThan = (currentServerVersion, oldServerVersion) => currentServerVersion && gte(coerce(currentServerVersion), oldServerVersion);
```
Comment (diegolmello) [inline app/lib/utils.js:15]: I have concerns about this naming still.
`server greater than` is actually `greater than or equal to`.
That might cause some issues in the future.
Do you agree?

What if we do something like this?
```js
export const methods = {
	lowerThan: lt,
	lowerThanOrEqualTo: lte,
	greaterThan: gt,
	greaterThanOrEqualTo: gte
};

export const compareServerVersion = (currentServerVersion, versionToCompare, func) => currentServerVersion && func(coerce(currentServerVersion), versionToCompare);
```

Usage would be like this:
```js
console.log('1.0.0 lower than 1.1.0', compareServerVersion('1.0.0', '1.1.0', methods.lowerThan));
console.log('1.0.0 lower than or equal to 1.1.0', compareServerVersion('1.0.0', '1.1.0', methods.lowerThanOrEqualTo));
console.log('1.0.0 greater than 1.1.0', compareServerVersion('1.0.0', '1.1.0', methods.greaterThan));
console.log('1.0.0 greater than or equal to 1.1.0', compareServerVersion('1.0.0', '1.1.0', methods.greaterThanOrEqualTo));
```

WDYT? @gerzonc

**Comment by diegolmello** [comment]
@gerzonc Lint is failing

**Review by diegolmello — APPROVED** [review]
Great job!

**Pull request merged**